pkg/routes/router_test.go

```diff
@@ -0,0 +1,268 @@
+package routes
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"mime/multipart"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/AngelVI13/fiber-investigation/pkg/database"
+	"github.com/AngelVI13/fiber-investigation/pkg/session"
+	"github.com/AngelVI13/fiber-investigation/pkg/testutil"
+	"github.com/gofiber/fiber/v2"
+)
+
+const (
+	FiberCookieName = "fiber-app-flash"
+)
+
+type testHandler func(app *fiber.App, t *testing.T)
+
+func NewTestRouter(t *testing.T) *Router {
+	db := testutil.NewTestDb(t)
+	return NewRouter(db)
+}
+
+func MultipartForm(data map[string]string) (
+	body *bytes.Buffer,
+	formType string,
+	err error,
+) {
+	// New multipart writer.
+	body = &bytes.Buffer{}
+	writer := multipart.NewWriter(body)
+	defer writer.Close()
+
+	for k, v := range data {
+		fw, err := writer.CreateFormField(k)
+		if err != nil {
+			return nil, "", fmt.Errorf(
+				"failed to create form field %s: %v", k, err,
+			)
+		}
+		_, err = io.Copy(fw, strings.NewReader(v))
+		if err != nil {
+			return nil, "", fmt.Errorf(
+				"failed to add value %s to form field %s: %v", v, k, err,
+			)
+		}
+	}
+
+	return body, writer.FormDataContentType(), nil
+}
+
+func CheckCookieStatus(resp *http.Response, status string, t *testing.T) {
+	cookies := resp.Cookies()
+	if len(cookies) < 1 {
+		t.Errorf("unexpected number of cookies: wanted 1 but got %d", len(cookies))
+	}
+
+	cookie := cookies[0]
+	if cookie.Name != FiberCookieName {
+		t.Errorf("unexpected cookied name: %s", cookie.Name)
+	}
+
+	if !strings.Contains(cookie.Value, status) {
+		t.Fatalf("expected %s to be flashed on screen but got: %s", status, cookie.Value)
```

Comment:
just for me to understand. You mentioned that you will double check if Errorf and FatalF are different for this testing approach.

Reply:
Error and Fatal seem to work as they're suppossed to